install-dev/upgrade/php/add_new_status_stock.php

```diff
@@ -24,67 +24,105 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
-/**
- * Select all current payment modules for the carrier restriction
- */
 function add_new_status_stock()
 {
     $translator = Context::getContext()->getTranslator();
+    $languages  = Language::getLanguages();
+
+    // insert ps_tab AdminStockManagement
+    $count = (int)Db::getInstance()->getValue(
+        'SELECT count(id_tab) FROM `' . _DB_PREFIX_ . 'tab` 
+        WHERE `class_name` = \'AdminStockManagement\'
+        AND `id_parent` = 9'
+    );
+    if (!$count) {
+        Db::getInstance()->execute(
+            'INSERT INTO `' . _DB_PREFIX_ . 'tab`
+            (`id_tab`, `id_parent`, `position`, `module`, `class_name`, `active`, `hide_host_mode`, `icon`)
+            VALUES (null, 9, 7, NULL, \'AdminStockManagement\', 1, 0, \'\')'
+        );
+        $lastIdTab = Db::getInstance()->insert_id();
 
+        // ps_tab_lang
+        $queryString = "INSERT INTO `%stab_lang` (`id_tab`, `id_lang`, `name`) VALUES (%s, %s, '%s')";
+        foreach ($languages as $lang) {
+            $stockName = pSQL(
+                $translator->trans(
+                    'Stock',
+                    array(),
+                    'Admin.Navigation.Menu',
+                    $lang['locale']
+                )
+            );
+            Db::getInstance()->execute(
+                sprintf(
```

Comment:
You should avoid the `sprintf()` calls. This may be more beautiful for you, but this seems to bring more long-term issues.
People usually see concatenation instead of sprintf(), and while a concatenation badly written can be detected immediately by the different IDEs, a `sprintf()` with the wrong number of parameters won't be detected before the execution. Also, it requires more CPU time.

The only good reason to use a `sprintf()` is when the parameter position may change, like in a string to translate.

Reply:
Got the point.

Reply:
Thanks!